lib/elixir/lib/protocol.ex

```diff
@@ -549,6 +479,10 @@ defmodule Protocol do
           nil
         end
 
+      # Disable dialyzer checks - before and after consolidation
+      # the types could be more strict
+      @dialyzer {:nowarn_function, __protocol__: 1, impl_for: 1, impl_for!: 1}
```

Comment:
I would like to keep the previous behaviour because `impl_for/1` and `impl_for/2` is used in applications and changing the specs on consolidation meant we could catch bugs that now we won't.

Reply:
Actually, ignore this. We were not rewriting `impl_for` specs and the `__protocol__` ones are not relevant to what I said above.